openmc/universe.py

```diff
@@ -600,3 +625,163 @@ def _determine_paths(self, path='', instances_only=False):
+    universe_id : int, optional
+        Unique identifier of the universe. If not specified, an identifier will
+        automatically be assigned.
+    name : str, optional
+        Name of the universe. If not specified, the name is the empty string.
+    auto_geom_ids : bool
+        Set IDs automatically on initialization (True) or report overlaps
+        in ID space between CSG and DAGMC (False)
+    auto_mat_ids : bool
+        Set IDs automatically on initialization (True)  or report overlaps
+        in ID space between OpenMC and UWUW materials (False)
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier of the universe
+    name : str
+        Name of the universe
+    filename : str
+        Path to the DAGMC file used to represent this universe.
+    auto_geom_ids : bool
+        Set IDs automatically on initialization (True) or report overlaps
+        in ID space between CSG and DAGMC (False)
+    auto_mat_ids : bool
+        Set IDs automatically on initialization (True)  or report overlaps
+        in ID space between OpenMC and UWUW materials (False)
+    """
+
+    def __init__(self,
+                 filename,
+                 universe_id=None,
+                 name='',
+                 auto_geom_ids=False,
+                 auto_mat_ids=False):
+        super().__init__(universe_id, name)
+        # Initialize class attributes
+        self.filename = filename
+        self.auto_geom_ids = auto_geom_ids
+        self.auto_mat_ids = auto_mat_ids
+
+    def __repr__(self):
+        string = super().__repr__()
+        string += '{: <16}=\t{}\n'.format('\tGeom', 'DAGMC')
+        string += '{: <16}=\t{}\n'.format('\tFile', self.filename)
+        return string
+
+    @property
+    def filename(self):
+        return self._filename
+
+    @filename.setter
+    def filename(self, val):
+        cv.check_type('DAGMC filename', val, str)
+        self._filename = val
+
+    @property
+    def auto_geom_ids(self):
+        return self._auto_geom_ids
+
+    @auto_geom_ids.setter
+    def auto_geom_ids(self, val):
+        cv.check_type('DAGMC automatic geometry ids', val, bool)
+        self._auto_geom_ids = val
+
+    @property
+    def auto_mat_ids(self):
+        return self._auto_mat_ids
+
+    @auto_mat_ids.setter
+    def auto_mat_ids(self, val):
+        cv.check_type('DAGMC automatic material ids', val, bool)
+        self._auto_mat_ids = val
+
+    def get_all_cells(self, memo=None):
+        return OrderedDict()
+
+    def get_all_materials(self, memo=None):
+        return OrderedDict()
+
+    def create_xml_subelement(self, xml_element, memo=None):
+        if memo and self in memo:
+            return
+
+        if memo is not None:
+            memo.add(self)
+
+        # Set xml element values
+        dagmc_element = ET.Element('dagmc_universe')
+        dagmc_element.set('id', str(self.id))
+        dagmc_element.set('name', self.name)
```

Comment:
Do we really want to allow these universes to be named? Right now, although you can give a regular universe a name, nothing is actually done with it since we don't write a `<universe>` element in the XML. At the very least, if the `name` is not assigned, we shouldn't create an empty `name=""` attribute in the XML.

Reply:
Good point. I was realizing this too looking at the import/export XML methods yesterday. I think it's probably best to make the behavior of the name attribute as similar to the regular universe as possible.

Reply:
One instance where it would be nice to have the universe name is when looking at different levels of the geometry in the plotter so we can display universe names in the status bar. I think it can be dealt with in a separate PR if we feel it's important enough to get those names into the XML files though.